src/token/erc20.sol

```diff
@@ -66,78 +57,186 @@ contract ERC20 {
         );
     }
 
-    // --- ERC20 ---
-    function transfer(address dst, uint256 wad) external returns (bool) {
-        return transferFrom(msg.sender, dst, wad);
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return block.chainid == deploymentChainId ? _DOMAIN_SEPARATOR : _calculateDomainSeparator(block.chainid);
+    }
```

Comment:
Not sure I follow this domain separator stuff? 🤔

Reply:
This is copied from the most up-to-date token implementation from Maker: https://github.com/makerdao/xdomain-dss/blob/master/src/Dai.sol I also have to dig a bit deeper to fully understand this still, but for now I'd say trusting the existing implementation is fine.